internal/resolution/client/maven_registry_client.go

```diff
@@ -106,6 +106,9 @@ func (c *MavenRegistryClient) Requirements(ctx context.Context, vk resolve.Versi
 	if err := proj.MergeProfiles("", maven.ActivationOS{}); err != nil {
 		return nil, err
 	}
+	for _, repo := range proj.Repositories {
+		c.api.Add(string(repo.URL))
+	}
```

Comment:
Should this be adding new repositories in `Requirements`? The seems like you'd inherit the repositories of your dependencies?

Reply:
hmmmm does this happen only for dependency resolution? I assume you may not know what repositories defined in dependencies if only preparing effective POM?

Reply:
discussed offline - a project does not inherit repositories of your dependencies so we shouldn't add new repositories here